app/src/main/java/ch/epfl/sweng/erpa/services/GCP/GCPGameService.java

```diff
@@ -0,0 +1,59 @@
+package ch.epfl.sweng.erpa.services.GCP;
+
+import com.annimon.stream.Optional;
+
+import java.io.IOException;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import ch.epfl.sweng.erpa.model.Game;
+import ch.epfl.sweng.erpa.model.ObservableList;
+import ch.epfl.sweng.erpa.model.UserProfile;
+import ch.epfl.sweng.erpa.services.GameService;
+
+@Singleton
+public class GCPGameService implements GameService {
+
+    private final GCPRemoteServicesProvider grsp;
+    private final GCPService.GameInterface gameInterface;
+
+    @Inject public GCPGameService(GCPRemoteServicesProvider grsp) {
+        this.grsp = grsp;
+        if (grsp.getRetrofit() != null)
+            this.gameInterface = grsp.getRetrofit().create(GCPService.GameInterface.class);
+        else
+            this.gameInterface = null;
```

Comment:
Is this compliant with the representation invariant?

Reply:
it's necessary for testing

Reply:
Though I might get rid of it, depending on how I refactor tests, it might no longer be necessary